.github/workflows/check-file-updates.yaml

```
@@ -0,0 +1,45 @@
+name: Check config and readme updates
+on:
+  pull_request:
+    paths:
+      - "apis/**"
+jobs:
+  file-updates:
+    permissions:
+      pull-requests: write
+    name: Test file updates
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run make commands
+        run: |
+          make generate && make manifests && make api-docs
+      - name: Check for unstaged files
+        id: check_unstaged_files
+        run : |
+          if [[ -n $(git status -s) ]]
+          then
+            echo "New files to commit available"
+            echo "is_files_to_commit_present=true" >> $GITHUB_OUTPUT
+          else
+            echo "No new files to commit"
+            echo "is_files_to_commit_present=false" >> $GITHUB_OUTPUT
+          fi
+      - name: Generate comments
+        if: ${{ steps.check_unstaged_files.outputs.is_files_to_commit_present == 'true' }}
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          message: |
+              ## Important Message :x:
+
+              This PR cannot be merged as some files are not upto date. Please run the below commands to auto-generate files and make sure to commit them:
+                - `make generate`
+                - `make manifests`
+                - `make api-docs`
```

Why not all at once?

```suggestion
                - `make generate manifests api-docs`
```